scene/3d/physics/soft_body_3d.cpp

```diff
@@ -78,8 +81,48 @@ void SoftBodyRenderingServerHandler::close() {
 	write_buffer = nullptr;
 }
 
-void SoftBodyRenderingServerHandler::commit_changes() {
-	RS::get_singleton()->mesh_surface_update_vertex_region(mesh, surface, 0, buffer);
+void SoftBodyRenderingServerHandler::fti_pump() {
+	buffer_prev = buffer;
+	aabb_prev = aabb;
+}
+
+void SoftBodyRenderingServerHandler::commit_changes(real_t p_interpolation_fraction) {
+	real_t f = p_interpolation_fraction;
+	Vector<uint8_t> buffer_interp = buffer;
+	AABB aabb_interp = aabb;
+
+	if (p_interpolation_fraction < 1) {
+		// AABB.
+		aabb_interp = AABB(aabb_prev.position.lerp(aabb.position, f), aabb_prev.size.lerp(aabb.size, f));
+
+		for (uint32_t i = 0; i < vertex_count; i++) {
+			// Vertex.
+			float *vertex_buffer_prev = reinterpret_cast<float *>(buffer_prev.ptrw() + i * stride + offset_vertices);
+			float *vertex_buffer_interp = reinterpret_cast<float *>(buffer_interp.ptrw() + i * stride + offset_vertices);
+
+			*vertex_buffer_interp = Math::lerp(*vertex_buffer_prev, *vertex_buffer_interp, (float)f);
+			vertex_buffer_interp++;
+			vertex_buffer_prev++;
+			*vertex_buffer_interp = Math::lerp(*vertex_buffer_prev, *vertex_buffer_interp, (float)f);
+			vertex_buffer_interp++;
+			vertex_buffer_prev++;
+			*vertex_buffer_interp = Math::lerp(*vertex_buffer_prev, *vertex_buffer_interp, (float)f);
+
+			// Normal.
+			uint32_t *normal_buffer_prev = reinterpret_cast<uint32_t *>(buffer_prev.ptrw() + i * normal_stride + offset_normal);
```

Comment:
If `buffer_prev` is `Vector`, we don't want to be calling `ptrw()` in a tight loop as it invokes the COW machinery.
Probably better to get raw pointers outside the loop?